magma/lib/magma/actions/add_project.rb

```diff
@@ -2,11 +2,22 @@ class Magma
 
   class AddProjectAction < ComposedAction
     def perform
-      begin
-        Magma.instance.db.run "CREATE SCHEMA #{@project_name}"
-      rescue Sequel::Error => e
-        raise unless e.message.start_with?('PG::DuplicateSchema: ERROR:')
-      end
+      # Unfortunately we cannot just run a create and catch the error due to the way ruby sequel poorly handles graceful
+      # recovery of error inside transaction, which this is run inside of implicitly from parent caller.
+      Magma.instance.db.run <<-SQL
+DO $$
```

Comment:
I also found that relying on catching the sequel here for idempotence doesn't actually work -- it results in errors and does not continue to adding the project model as it should.  This has to do with the way sequel handles errors in transactions (it basically refuses to gracefully recover until the transaction is restarted, but the transaction is owned by the update_model controller and notthis level of code).

Solution was to use some PSQL here to check for the existence of the schema before executing the create.